### DISCUSSION TITLE
PIP package

### DISCUSSION THREAD

**Discussion by fakerybakery** [yl4579/StyleTTS2#93] [General]
Hi, are you planning to allow us to install this via pip by creating a `setup.py` file?

**Comment by yl4579** [comment]
This repo serves as a research's project, but if it needed surely I can make a pip package. Do you need it for training or for inference?

**Comment by fakerybakery** [comment]
Mostly for inference. I was planning to make a Gradio demo, but I can also do that by just cloning the repo. (Might make a PR later if that's ok.)

**Comment by yl4579** [comment]
Yes, please do so if you can. Thanks for your contribution.

**Comment by fakerybakery** [comment]
This will require a restructuring of the files (putting them all in one folder) and may break the demos, so I'll work on it in a fork for now.

**Comment by fakerybakery** [comment]
I've got a _very_ hacky fork that should be pip installable [here](https://github.com/fakerybakery/StyleTTS2) (still need to create `setup.py`), however the training scripts + config files have been removed.

**Comment by lxe** [comment]
Thanks for this, and the demo, @fakerybakery

Would be nice to have inference.py commited and mainlined.

**Comment by fakerybakery** [comment]
Hi @lxe, by `inference.py` do you mean a importable script? I'm planning to make a PR to merge the demo + script soon after I clean up the code.

**I'm going to delete my pip fork because it breaks training, config, etc.** Personally, I think it would be better to have an external "wrapper" repository that's pip-installable to keep the training scripts, similar to how VoiceFixer has the VoiceFixer pip installable package and the VoiceFixer main repository with training scripts, etc.

**Comment by fakerybakery** [comment]
@yl4579 would you mind adding this to the "todo" list in the README?

**Comment by yl4579** [comment]
My main concern of this is it could contain phonemizer and change the license of the repo. I use notebook intentionally so it doesn't contain any non-MIT libraries. Even if I add it to TODO it probably wouldn't be a part of this repo anyway.

**Comment by fakerybakery** [comment]
OK, please see the other comment about the fork

**Comment by lxe** [comment]
I was able to wrangle around the inability to import this in my own package:

https://github.com/lxe/tts-server

Here's what I needed to do:
 - styletts2 is a git submodule
 - needed to add this to make some submodules work https://github.com/lxe/tts-server/blob/4903b4a14828549b78676cb9927e0242695d5a06/setup.py#L4-L6
 - also needed this to make other impotrs inside styletts work: https://github.com/lxe/tts-server/blob/4903b4a14828549b78676cb9927e0242695d5a06/tts_server/tts.py#L20-L23
 
 I think a lot of this styletts code needs some relative module imports and some other cleanup

**Comment by yl4579** [comment]
I think I can move this to discussion now, as making a pip package would inevitably ruin the current MIT license status unless we have a good MIT licensed phonemizer alternative.
